### PR TITLE
start best-practices section

### PR DESCRIPTION
**Pull request by natebrennand**

### CODE REVIEW

**Comment by natebrennand** [comment]
CC @peggyl @cgclever @azylman

File: golang/README.md [hunk 1]
```diff
@@ -1,15 +1,65 @@
...
-All golang services (with some exceptions, like Heka) at Clever should be on go-1.5.
+All golang services written by Clever should be on Go 1.5.
+Some 3rd-party tools that we use, such as Heka, are not on Go 1.5.
+
+
+## Best Practices
+
+### Sets
+
+Sets are typically best represented by a `map[key type]bool` structure where every boolean value is set to `true`.
```
Comment (agclever) [inline golang/README.md]: What about using `map[key type]struct{}`?

Comment (natebrennand) [inline golang/README.md]: So if we're in a memory constrained environment the merit of using [an empty struct](http://dave.cheney.net/2014/03/25/the-empty-struct) is obviously the right move, but that is never really our case. IMO a comment explaining how the map is meant to be used is worth the improvement in dev UX.

I think the benefit is much more tangible in an `if` case where there is > 1 values being checked. Here's an example from Aeris: https://github.com/Clever/aeries/pull/108/files#diff-b06bec158b47475dafd35faa56ca4980R243

previous `if _, isException := httpsExceptions[system.ID]; parsedURL.Scheme == "http" && !isException {	`
updated: `if parsedURL.Scheme == "http" && !httpsExceptions[system.ID] {`

(in retrospect renaming `httpsExceptions` to `isHTTPSException` would've been a good move too)

Comment (peggyl) [inline golang/README.md]: I've seen this library before: https://github.com/fatih/set

IIRC, it uses an underlying `map[interface{}]struct{}` as well, but also provides a nice abstraction overall if we specifically need a set structure for a larger use case.

Comment (natebrennand) [inline golang/README.md]: Ya, in a case where you need more than just insertion and presence using a full featured implementation is definitely a good move :+1:

**Pull request merged**